I'm not sure if this is necessary, but the original driver did it and
apparently also many other drivers do it, so it should not be completely
wrong to do it.

Signed-off-by: Phil Sutter <n0-1@freewrt.org>
---
 drivers/net/korina.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

--- a/drivers/net/korina.c
+++ b/drivers/net/korina.c
@@ -418,6 +418,7 @@ static int korina_rx(struct net_device *
 			/* 16 bit align */
 			skb_reserve(skb_new, 2);
 
+			skb_new->dev = dev;
 			lp->rx_skb[lp->rx_next_done] = skb_new;
 		}
 		rd->devcs = 0;
@@ -761,6 +762,7 @@ static void korina_alloc_ring(struct net
 		if (!skb)
 			break;
 		skb_reserve(skb, 2);
+		skb->dev = dev;
 		lp->rx_skb[i] = skb;
 		lp->rd_ring[i].control = DMA_DESC_IOD |
 				DMA_COUNT(KORINA_RBSIZE);
